services/rpz.c

```diff
@@ -427,16 +439,22 @@ rpz_create(struct config_auth* p)
 	return NULL;
 }
 
-/** Remove RPZ zone name from dname */
+/**
+ * Remove RPZ zone name from dname
+ * Copy dname to newdname, without the originlen number of trailing bytes
+ */
 static size_t
 strip_dname_origin(uint8_t* dname, size_t dnamelen, size_t originlen,
-	uint8_t* newdname)
+	uint8_t* newdname, size_t maxnewdnamelen)
 {
 	size_t newdnamelen;
 	if(dnamelen < originlen)
 		return 0;
 	newdnamelen = dnamelen - originlen;
+	if(newdnamelen+1 > maxnewdnamelen)
+		return 0;
 	memmove(newdname, dname, newdnamelen);
+	memset(newdname+newdnamelen, 0, 1);
```

Comment:
This could be `newdname[newdnamelen]=0;` but this memset works as well.

Reply:
Agree, that might be more easy to read.